samples/cpp/visual_language_chat/benchmark_vlm.cpp

```diff
@@ -96,6 +100,7 @@ int main(int argc, char* argv[]) try {
     std::cout << "TPOT: " << metrics.get_tpot().mean  << " ± " << metrics.get_tpot().std << " ms/token " << std::endl;
     std::cout << "Throughput: " << metrics.get_throughput().mean  << " ± " << metrics.get_throughput().std << " tokens/s" << std::endl;
 
+    delete pipe;
```

Comment:
Does `std::optional` work here instead of `VLMPipeline*`? If so, use it. If not, use `std::unique_ptr` so there's no need to care about freeing memory if VLMPipeline throws